src/handler/archive.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { validateArchiveErrors } from '../validators/archive';
+import { addHttpHeaders } from '../util/httpHeaders';
+import { archiveRecord, getBySystemNumberAndCreatedTimestamp } from '../services/database';
+import { Status } from '../util/enum';
+import { getUserDetails } from '../services/user';
+import { formatTechRecord } from '../util/formatTechRecord';
+import { ArchiveRecord, ArchiveRecordRequestBody } from '../models/archive';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Archive end point called');
+
+  const archiveErrors = validateArchiveErrors(event);
+
+  if (archiveErrors) {
+    return addHttpHeaders(archiveErrors);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const body: ArchiveRecordRequestBody = await JSON.parse(event.body as string);
+
+  if (!body.reasonForArchiving) {
+    return {
+      statusCode: 400,
+      body: 'Reason for archiving not provided',
+    };
+  }
+
+  const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber as string);
+  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp as string);
+  const userDetails = getUserDetails(event.headers.Authorization as string);
+
+  logger.info(`Get from database with sysNum ${systemNumber} and timestamp ${createdTimestamp}`);
+
+  const record = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp) as ArchiveRecord;
+
+  logger.debug(`result is: ${JSON.stringify(record)}`);
+
+  if (!record || !Object.keys(record).length) {
```

Comment:
Maybe the method from processPostRequest could be moved into a shared validator as the logic is very similar?